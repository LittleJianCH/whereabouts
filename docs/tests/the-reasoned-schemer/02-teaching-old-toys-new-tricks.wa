Car [list, car]
--------------- {
  unify [car, ...cdr] = list
}

query q {
  Car [["a", "c", "o", "r", "n"], q]
}

query q {
  Car [["a", "c", "o", "r", "n"], "a"]
}

query r {
  Car [[r, ...y], x]
  unify "pear" = x
}
