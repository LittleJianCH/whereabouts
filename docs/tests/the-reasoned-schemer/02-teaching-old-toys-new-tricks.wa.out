["a"]
["?q"]
["pear"]
[["grape", "a"]]
["o"]
[[["raisin", "pear"], "a"]]
["?q"]
["o"]
[["a", "c", "o", "r", "n"]]
[[["a", "b", "c"], "d", "e"]]
["d"]
[["e", "a", "d", "c"]]
["d"]
[["d", "a", "d", "c"]]
[["d", "a", "d", "c"]]
